2D Simulation/movesCorrector.js

```diff
@@ -0,0 +1,98 @@
+let movesCorrector = new function () {
+    let undo_action;
+    let incorrect_button = $("#buttonIncorrect");
+    let awaiting_correction = false;
+
+    incorrect_button.prop("disabled", true);
+
+    this.handle_incorrect_move = function () {
+        awaiting_correction = true;
+
+        this.disable_incorrect_button();
+
+        run_undo_action();
+        display_block_ids();
+        display_flip_explanation();
+    }
+
+    function run_undo_action() {
+        if (undo_action === undefined) {
+            return;
+        }
+
+        if (undo_action.type === "move") {
+            update_position(undo_action);
+            update_score(undo_action);
+        } else if (undo_action.type === "flip") {
+            flipBlock(undo_action.block_id, null, currentConfig);
+        }
+
+        undo_action = undefined;
+    }
+
+    function display_block_ids() {
+        for (let i = 0; i < NumBlocks; i++) {
+            blocks.set_block_text(i, i);
+        }
+    }
+
+    function display_flip_explanation() {
+        alert("Please enter the id number of the block you wanted to flip.");
+    }
+
+    this.disable_incorrect_button = function () {
+        incorrect_button.prop("disabled", true);
+    };
+
+    this.enable_incorrect_button = function () {
+        incorrect_button.prop("disabled", false);
+        hide_block_ids();
+    };
+
+    function hide_block_ids() {
```

Comment:
It seems like a more natural name for this would be something like `display_block_letters`. Any particular reason you chose what you have here?

Reply:
It was supposed to be the inverse of `display_block_ids`, but I like yours better.